ts/client/services/syncer.ts

```diff
@@ -16,28 +14,19 @@ export enum SyncMode {
 
 export class SyncerService {
     private mode: SyncMode;
-    private newBatchFilter: EventFilter;
-    private singlePubkeyRegisteredFilter: EventFilter;
-    private batchPubkeyRegisteredFilter: EventFilter;
-    private batchHandlingContext: BatchHandlingContext;
-    private rollup: Rollup;
-    private accountRegistry: BlsAccountRegistry;
-    private pubkeyStorage: PubkeyStorageEngine;
+    private readonly events: EventSyncer;
 
     constructor(private readonly api: CoreAPI) {
         this.mode = SyncMode.INITIAL_SYNCING;
-        this.rollup = this.api.rollup;
-        this.accountRegistry = this.api.contracts.blsAccountRegistry;
-        this.newBatchFilter = this.rollup.filters.NewBatch(null, null, null);
-        this.singlePubkeyRegisteredFilter = this.accountRegistry.filters.SinglePubkeyRegistered(
-            null
-        );
-        this.batchPubkeyRegisteredFilter = this.accountRegistry.filters.BatchPubkeyRegistered(
-            null,
-            null
-        );
-        this.batchHandlingContext = new BatchHandlingContext(api);
-        this.pubkeyStorage = this.api.l2Storage.pubkey;
+
+        this.events = new SequentialCompositeEventSyncer([
+            // Note: Ordering here is important for initial syncs.
+            // Pubkey syncs need to happen before batch syncs, etc.
+            new SinglePubkeyRegisteredEventSyncer(api),
+            new BatchPubkeyRegisteredEventSyncer(api),
+            new DepositQueuedEventSyncer(api),
+            new NewBatchEventSyncer(api)
```

Comment:
No action to take here. I'm commenting on why in that 100 eth1 blocks interval, we can process all the deposit subtrees first then process all the deposit batches later.

Say if these events happen in a sequence in that interval:
- subtree1 enqueued
- subtree2 enqueued
- subtree1 dequeued and submitted to Rollup.sol
-  subtree3 enqueued
-  subtree2 dequeued and submitted to Rollup.sol

Since deposit subtrees are in a queue, the Rollup.sol always consumes from the front and new deposits are always added from the back.

Reply:
@ChihChengLiang Would this be good to document in this code in the future to provide more context around the order?

Reply:
Sounds like a good idea